### PR TITLE
Add a "IREE developer tips and tricks" guide.

### PR DESCRIPTION
**Pull request by ScottTodd**
This is a collection of tips that I think will be useful to enough "developers" or "curious users" to include on the website.

I took some ideas from https://github.com/openxla/iree/issues/7116 while writing this page - I'm generally trying to demystify how the compiler works and what artifacts it produces.

Preview: https://scotttodd.github.io/iree/guides/developer-tips/

### CODE REVIEW

File: docs/website/docs/guides/developer-tips.md [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+
+    !!! tip "Tip - Embedded and system linking"
+
+        The default value of `--iree-llvmcpu-link-embedded=true` generates
+        platform-agnostic ELF files. By disabling that flag, the compiler will
+        produce `.so` files for Linux, `.dll` files for Windows, etc. While ELF
+        files are more portable, inspection of compiled artifacts is easier with
+        platform-specific shared object files.
+
+    ??? tip "Tip - Disassembling `.bc` files with `llvm-dis`"
```
Comment (ScottTodd) [inline docs/website/docs/guides/developer-tips.md:181]: @bjacob or @MaheshRavishankar - any feedback on these tips? (I've seen you mention using `llvm-dis` in the context of explaining how codegen works, and I'd hope that we could point people at this page to get them started self-service)

Comment (bjacob) [reply docs/website/docs/guides/developer-tips.md:181]: looks perfect!

**Review by bjacob — APPROVED** [review]

File: docs/website/docs/guides/developer-tips.md (hunk 1 above)
Comment (bjacob) [reply docs/website/docs/guides/developer-tips.md:181]: ok, one suggestion, I would mention that `llvm-dis` can easily be built in a IREE build directory:

```
benoitjacob@cloud:~/iree-build-linux$ ninja llvm-dis
[0/2] Re-checking globbed directories...
[2/2] Linking CXX executable llvm-project/bin/llvm-dis
```

File: docs/website/docs/guides/developer-tips.md [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+    simple_abs_cpu.vmfb
+    ```
+
+    !!! tip "Tip - Embedded and system linking"
+
+        The default value of `--iree-llvmcpu-link-embedded=true` generates
+        platform-agnostic ELF files. By disabling that flag, the compiler will
+        produce `.so` files for Linux, `.dll` files for Windows, etc. While ELF
+        files are more portable, inspection of compiled artifacts is easier with
+        platform-specific shared object files.
```
Comment (bjacob) [inline docs/website/docs/guides/developer-tips.md]: I would rephrase this slightly to be less about ELF vs non-ELF container format, and more about platform-standard vs IREE-custom. Because even on platforms like Linux where ELF is standard, IREE's flavor (embedded-elf) is not as thoroughly understood by non-IREE tools. Probably just as a matter of being more aggressively optimized for size. IREE says "symbol table? who needs a symbol table when you have `iree_hal_executable_library_query` ? Of course, a symbol table is useful to use tools such as `objdump` to disassemble symbols, when using profilers, Tracy sampling, etc.

File: docs/website/docs/guides/developer-tips.md (hunk 1 above)
Comment (ScottTodd) [reply docs/website/docs/guides/developer-tips.md:181]: Good idea, added a brief section of the code blocks to show how to build.

File: docs/website/docs/guides/developer-tips.md (hunk 2 above)
Comment (ScottTodd) [reply docs/website/docs/guides/developer-tips.md]: Okay, reworded.

File: docs/website/docs/guides/developer-tips.md [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+  }
+}
+```
+
+This is similar to the `--mlir-print-ir-after=` flag, but at clearly defined
+pipeline phases.
+
+Similarly, compilation can be continued from any intermediate phase. This allows
+for interative workflows - compile to a phase, make edits to the `.mlir` file,
+then resume compilation and continue through the pipeline.
```
Comment (kuhar) [inline docs/website/docs/guides/developer-tips.md]: This is a great piece of advice! Could we also show the command to resume compilation from some `.mlir`'?

**Review by kuhar — APPROVED** [review]
This is fantastic, thanks for collecting these tips!

I left one suggestion.

File: docs/website/docs/guides/developer-tips.md (hunk 3 above)
Comment (ScottTodd) [reply docs/website/docs/guides/developer-tips.md]: Sure thing. Added a section. Here's how it renders: 
image

(is changing `absf` to `exp` a useful/clear enough example?)

Comment (ScottTodd) [reply docs/website/docs/guides/developer-tips.md]: BTW, these test files show other ways of using those features (but I don't really feel like linking them here... could be persuaded):

* https://github.com/openxla/iree/blob/main/tools/test/compile_to_phase.mlir
* https://github.com/openxla/iree/blob/main/tools/test/compile_to_continuation.mlir

Also see

* https://github.com/openxla/iree/blob/main/tools/test/compile_pipelines.mlir
